packages/template-retail-react-app/app/pages/checkout/partials/shipping-address-selection.jsx

```diff
@@ -105,11 +105,12 @@ const ShippingAddressSelection = ({
     onSubmit = async () => null
 }) => {
     const {formatMessage} = useIntl()
-    const {data: customer} = useCurrentCustomer()
+    const {data: customer, isLoading, isFetching} = useCurrentCustomer()
+    const isLoadingRegisteredCustomer = isLoading && isFetching
 
-    const hasSavedAddresses = customer.addresses && customer.addresses.length > 0
-    const [isEditingAddress, setIsEditingAddress] = useState(!hasSavedAddresses)
-    const [selectedAddressId, setSelectedAddressId] = useState(false)
+    const hasSavedAddresses = customer.addresses?.length > 0
+    const [isEditingAddress, setIsEditingAddress] = useState(false)
+    const [selectedAddressId, setSelectedAddressId] = useState(undefined)
```

Comment:
The changes in this file are to address the scenario when the current customer data is not fully available yet. This scenario leads to an intermittent bug that I've logged in GUS ticket W-12779702.

Previously it was hard to reproduce the bug locally. But now the checkout tests can reproduce it consistently.